Sources/SwiftDocC/Model/Rendering/Navigation Tree/RenderHierarchyTranslator.swift

```diff
@@ -192,19 +192,42 @@ struct RenderHierarchyTranslator {
     /// The documentation model is a graph (and not a tree) so you can curate API symbols
     /// multiple times under other API symbols, articles, or API collections. This method
     /// returns all the paths (breadcrumbs) between the framework landing page and the given symbol.
-    mutating func visitSymbol(_ symbolReference: ResolvedTopicReference) -> RenderHierarchy {
-        let pathReferences = context.finitePaths(to: symbolReference)
-        pathReferences.forEach({
-            collectedTopicReferences.formUnion($0)
-        })
-        let paths = pathReferences.map { $0.map { $0.absoluteString } }
-        return .reference(RenderReferenceHierarchy(paths: paths))
+    mutating func visitSymbol(_ symbolReference: ResolvedTopicReference) -> VariantCollection<RenderHierarchy?> {
+        // An inner function used to create a RenderHierarchy from a list of references and collect the unique references
+        func makeHierarchy(_ references: [ResolvedTopicReference]) -> RenderHierarchy {
+            collectedTopicReferences.formUnion(references)
+            let paths = references.map(\.absoluteString)
+            return .reference(.init(paths: [paths]))
+        }
+        
+        let mainPathReferences = context.linkResolver.localResolver.breadcrumbs(of: symbolReference, in: symbolReference.sourceLanguage)
+        
+        var hierarchyVariants = VariantCollection<RenderHierarchy?>(
+            defaultValue: mainPathReferences.map(makeHierarchy) // It's possible that the symbol only has a language representation in a variant language
+        )
+        
+        for language in symbolReference.sourceLanguages where language != symbolReference.sourceLanguage {
+            guard let variantPathReferences = context.linkResolver.localResolver.breadcrumbs(of: symbolReference, in: language) else {
+                continue
+            }
+            hierarchyVariants.variants.append(.init(
+                traits: [.interfaceLanguage(language.id)],
+                patch: [.replace(value: makeHierarchy(variantPathReferences))]
+            ))
+        }
+        
+        return hierarchyVariants
     }
     
     /// Returns the hierarchy under a given article.
     /// - Parameter symbolReference: The reference to the article.
     /// - Returns: The framework hierarchy that describes all paths where the article is curated.
-    mutating func visitArticle(_ symbolReference: ResolvedTopicReference) -> RenderHierarchy {
-        return visitSymbol(symbolReference)
+    mutating func visitArticle(_ symbolReference: ResolvedTopicReference) -> VariantCollection<RenderHierarchy?> {
```

Comment:
Why is this function's parameter called `symbolReference` and not `articleReference` ?

Reply:
I didn't change it but I suspect that it was an old copy-and-paste mistake. I can fix it.

Reply:
Renamed in https://github.com/swiftlang/swift-docc/pull/1081/commits/e63244e65fe42acfc0b0b41f9c3aa8f7101f5396